### PR TITLE
update docs for for loop with conditional to reference filter filters

### PR DESCRIPTION
**Pull request by shieldo**
I don't know if there needs to be the same emphasis for this case as there's no special syntax for this any more? This PR assumes the same emphasis.

### CODE REVIEW

File: doc/tags/for.rst [hunk 1]
```diff
@@ -85,13 +85,14 @@ Adding a condition
...
-items. The following example skips all the users which are not active:
+can however apply a :doc:`filter<filters/filter>` filter onto the sequence
+which allows you, effectively, to skip items. The following example skips all
+the users which are not active:
 
 .. code-block:: twig
 
     <ul>
-        {% for user in users if user.active %}
+        {% for user in users|filter(v => v.active) %}
```
Comment (fabpot) [inline doc/tags/for.rst]: I would use `user` or `u` for the variable.

**Comment by fabpot** [comment]
Can you rebase on 1.x?

File: doc/tags/for.rst (hunk 1 above)
Comment (fabpot) [reply doc/tags/for.rst]: You should also add a note to say that `filter` is available as of Twig 1.41/2.10 (and mention the old syntax with the `if` condition).

**Comment by shieldo** [comment]
@fabpot Might it be better for me to submit another PR for the 1.x branch so I can reference 1.x releases there and 2.x releases here? I'm just looking at e.g. https://twig.symfony.com/doc/2.x/filters/filter.html#filter and it seems confusing/ possibly misleading that only a 1.x release is mentioned there in the 2.x docs (as the `filter` filter wasn't available in older 2.x releases).

**Comment by fabpot** [comment]
You're right

**Comment by shieldo** [comment]
@fabpot OK I changed my approach here and just added a tip for the deprecation/ replacement syntax, given it maybe doesn't strictly need a syntax example, and seemed more consistent with docs elsewhere (such as https://twig.symfony.com/doc/2.x/tags/spaceless.html). If that makes sense then I can submit a separate  PR for 1.x, and add a 3.x one that uses `filter` only.

File: doc/tags/for.rst [hunk 2]
```diff
@@ -84,6 +84,13 @@ Variable              Description
 Adding a condition
 ------------------
 
+.. tip::
+
+    As of Twig 2.10, use the :doc:`filter <../filters/filter>` filter instead,
+    or an ``if`` condition inside the ``for`` body (if your condition
+    depends on a variable updated inside the loop and you are not using the
+    `loop` variable).
```
Comment (fabpot) [inline doc/tags/for.rst]: double tick

**Comment by fabpot** [comment]
@shieldo Works for me.

File: doc/tags/for.rst (hunk 2 above)
Comment (shieldo) [reply doc/tags/for.rst]: Updated

**Comment by fabpot** [comment]
Thank you @shieldo.

**Pull request merged**